components/mediators/cache/org.wso2.carbon.mediator.cache/src/main/java/org/wso2/carbon/mediator/cache/digest/HttpRequestHashGenerator.java

```diff
@@ -76,10 +80,21 @@ public String getDigest(MessageContext msgContext) throws CachingException {
         //If some or all headers need to be included in the hash
         if (!excludeAllHeaders) {
             Map<String, String> transportHeaders =
-                    (Map<String, String>) msgContext.getProperty(MessageContext.TRANSPORT_HEADERS);
+                    new TreeMap<String, String>(new Comparator<String>() {
+                        public int compare(String o1, String o2) {
+                            return o1.compareToIgnoreCase(o2);
```

Comment:
why we are adding this line? Could you please explain the logic ( to compare the strings )

Reply:
According to RFC2626 http headers are case insensitive. Since I am creating a new map here, this compare method has to be defined.